vignettes/articles/cran-compliance.Rmd

```diff
@@ -0,0 +1,74 @@
+
+CRAN expects that if cargo is not on the `PATH`, the user's home directory is checked at `~/.cargo/bin`. The configuration files must perform these checks. 
+
+## `cargo build` settings 
+
+CRAN also places restrictions on the way that cargo builds. CRAN has requested that no more than two logical CPUs be used in the build process. By default, cargo uses multiple threads to speed up the compilation process. CRAN policy allows for a maximum of two. This is set using the `-j 2` option in cargo build. 
+
+Additionally, to minimize security risks and ensure package stability, CRAN requires that packages be built completely offline. This prevents external dependencies from being downloaded at compile time. Because of this requirement, vendored dependencies must be used.
+
+## Vendored dependencies 
+
+Vendoring dependencies is the act of including the dependency itself into a packages source code. In the case of Rust, dependencies are fetched only at compile time. To enable compilation in an offline environment, dependencies must be vendored which is accomplished using the `cargo vendor` command. 
+
+`cargo vendor` creates a local directory, called `vendor/` by default, which contains the source code for each of the recursive dependencies of the crate that is being built. For CRAN compatibility, the `vendor/` directory must be compressed using tar xz compression and included in the source of the package. 
+
+During build time, the dependencies are uncompressed, compiled in the offline build, and discarded. This process is controlled by the `Makevars` and `Makevars.win` files. 
+
+## Package compilation
+
+All of this comes together during package compilation time. During package compilation certain behaviors must be achieved: 
+
+- cargo must be able to be called from a user's home directory
+- the user's home directory must not be modified or written to 
+- the package must be compiled offline 
+- no more than two logical CPUs are used
+- the versions of `cargo` and `rustc` are printed
+
+
+## Using CRAN defaults
+
+rextendr provides default CRAN compliant scaffolding via the `use_cran_defaults()` function and appropriate vendoring with `vendor_pkgs()`.
+
+### Making a package CRAN compliant
+
+To create a CRAN compliant R package begin by creating a new R package. Do so by using `usethis::create_package()`. In the new R project, run `rextendr::use_extendr()` to create the minimal scaffolding necessary for a Rust powered R package. Once you have done this, you can now run `rextendr::use_cran_defaults()`. 
+
+`use_cran_defaults()` will create the files `configure` and `configure.win` files. Additionally, it will create new `Makevars` and `Makevars.win` that print the versions of cargo and rustc as well as uses the `cargo build` argument `-j 2 --offline`.
```

Comment:
```suggestion
`use_cran_defaults()` will create the `configure` and `configure.win` files. Additionally, it will create new `Makevars` and `Makevars.win` that print the versions of `cargo` and `rustc` as well as use the `cargo build` argument `-j 2 --offline`.
```